centreon/src/Core/MonitoringServer/Model/MonitoringServer.php

```diff
@@ -46,4 +60,61 @@ public function getName(): string
     {
         return $this->name;
     }
+
+    public function getEngineStartCommand(): ?string
+    {
+        return $this->engineStartCommand;
+    }
+
+    public function getEngineStopCommand(): ?string
+    {
+        return $this->engineStopCommand;
+    }
+
+    public function getEngineRestartCommand(): ?string
+    {
+        return $this->engineRestartCommand;
+    }
+
+    public function getEngineReloadCommand(): ?string
+    {
+        return $this->engineReloadCommand;
+    }
+
+    public function getBrokerReloadCommand(): ?string
+    {
+        return $this->brokerReloadCommand;
+    }
+
+    public function update(
+        string $name,
+        ?string $engineStartCommand,
+        ?string $engineStopCommand,
+        ?string $engineReloadCommand,
+        ?string $engineRestartCommand,
+        ?string $brokerReloadCommand
+    ) {
+        Assertion::notEmpty($name);
+        if ($engineStartCommand !== null) {
+            Assertion::regex($engineStartCommand, self::VALID_COMMAND_START_REGEX);
+        }
+        if ($engineStopCommand !== null) {
+            Assertion::regex($engineStopCommand, self::VALID_COMMAND_STOP_REGEX);
+        }
+        if ($engineReloadCommand !== null) {
+            Assertion::regex($engineReloadCommand, self::VALID_COMMAND_RELOAD_REGEX);
+        }
+        if ($engineRestartCommand !== null) {
+            Assertion::regex($engineRestartCommand, self::VALID_COMMAND_RESTART_REGEX);
+        }
+        if ($brokerReloadCommand !== null) {
+            Assertion::regex($brokerReloadCommand, self::VALID_COMMAND_RELOAD_REGEX);
+        }
```

Comment:
https://github.com/beberlei/assert?tab=readme-ov-file#nullor-helper

```suggestion
        Assertion::nullOrRegex($engineStartCommand, self::VALID_COMMAND_START_REGEX);
        Assertion::nullOrRegex($engineStopCommand, self::VALID_COMMAND_STOP_REGEX);
        Assertion::nullOrRegex($engineReloadCommand, self::VALID_COMMAND_RELOAD_REGEX);
        Assertion::nullOrRegex($engineRestartCommand, self::VALID_COMMAND_RESTART_REGEX);
        Assertion::nullOrRegex($brokerReloadCommand, self::VALID_COMMAND_RELOAD_REGEX);
```